composer/metrics/glue.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""A collection of classes to store GLUE metrics during finetuning using the NLP entrypoint. """
+
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+__all__ = ['GlueState', 'GLUEMetricsLogger']
+
+@dataclass
+class GLUEMetricsLogger:
+    """ Class mapping all GLUE tasks to their respective average metric values. 
+
+    Args:
+        tasks_to_avg_metric dict(str, Any): dictionary mapping GLUE task names to their avg values.
+    
+    """
+    task_to_avg_metric: Dict[str, Any]  
+
+    def __init__(self, task_names) -> None:
```

Comment:
When using a dataclass, no need to define `__init__` -- instead I'd make this a normal class, or use `__post_init__`.